src/mscorlib/shared/System/StringComparer.cs

```diff
@@ -170,6 +171,7 @@ public int GetHashCode(object obj)
     }
 
     [Serializable]
+    [System.Runtime.CompilerServices.TypeForwardedFrom("mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089")]
     internal sealed class CultureAwareComparer : StringComparer
```

Comment:
If this is something we need to do then we are going to need to make a number of changes to support it. We are going to need to start manually creating and maintaining an mscorlib ref in corefx that contains these internal types. We are also going to need to give InternalsVisibileTo access to mscorlib from System.Private.CoreLib. With those changes we can technically support this but I would like to avoid doing that if at all possible.